src/Portal.tsx

```diff
@@ -1,12 +1,14 @@
-import { PureComponent } from 'react';
+import * as React from 'react';
 import * as ReactDOM from 'react-dom';
+import { ThemeConsumer } from 'styled-components';
+import { IThemeInterface } from './types';
 
 export interface IPortalProps {
-  children: JSX.Element | string | number;
+  children: any;
   className?: string;
 }
 
-export class Portal extends PureComponent<IPortalProps> {
+export class Portal extends React.PureComponent<IPortalProps> {
   private readonly el?: HTMLDivElement;
   private readonly root = typeof document === 'object' ? document.body : null;
```

Comment:
I am not quite sure if we should attach Portal under document.body. Basically, this leads to the situation when theming is slightly broken, since we cannot rely on rules inheritance and we have to style components on our own (see a related story for a reference)

thoughts? @marbemac @lottamus @casserni

Reply:
I'm not sure that rules inheritance is applicable to styled components. Usually it adds classes to every component with a proper set of rules. Theming works by overriding some of the rules for every component.

Reply:
In general you are right, but unfortunately, some rules are (or might be) inherited (depending on global styles).
For instance, in this particular case, the color is inherited.
![image](https://user-images.githubusercontent.com/9273484/48546853-a343cc00-e8c9-11e8-9e4c-7255839baa90.png)
This is most certainly caused by the fact Box doesn't set color on its own node, but still - a developer might want to have some global styles for everything under given node.

Reply:
Ah, interesting! There are tradeoffs here, but for simplicity and for now I think the portal just needs to be within the theme provider (and thus have access to the theme). IE if you create a popup with a `<Button bg="error">Foo</Button>` in it, that button should have a red background when rendered in the portal. Does that make sense?

Reply:
Well, I added ThemeConsumer to the Portal, so its child components do have access to the theme. I think it's a nice tradeoff - you can still use theming, but also opt-out if you need to for some reason.
You can see that in action in the story book :)

Reply:
My apologies for dropping a ball here.
I adjusted the stories to show that the rendered content can be actually themed in the way @marbemac suggested.

I also made the final API a bit more end-user friendly, as I wrap the trigger component with required event handlers.